File: server/HabitTracker.Api/DTOs/AttributeOverviewResponse.cs
namespace HabitTracker.Api.DTOs;

public sealed class AttributeOverviewResponse
{
    public IReadOnlyList<UserAttributeResponse> Attributes { get; set; }
        = [];

    public int TotalAttributeXp { get; set; }

    public int BalanceScore { get; set; }

    public UserAttributeResponse? StrongestAttribute { get; set; }

    public UserAttributeResponse? NeedsFocusAttribute { get; set; }

    public IReadOnlyList<AttributeLevelUpResponse> ClosestToLevelUp
    { get; set; }
        = [];

    public IReadOnlyList<XpTransactionResponse> RecentXpTransactions
    { get; set; }
        = [];
}

File: client/src/types/AttributeOverviewResponse.ts
import type { AttributeLevelUpResponse } from './AttributeLevelUpResponse'
import type { UserAttributeResponse } from './UserAttributeResponse'
import type { XpTransactionResponse } from './XpTransactionResponse'

export type AttributeOverviewResponse = {
  attributes: UserAttributeResponse[]
  totalAttributeXp: number
  balanceScore: number
  strongestAttribute: UserAttributeResponse | null
  needsFocusAttribute: UserAttributeResponse | null
  closestToLevelUp: AttributeLevelUpResponse[]
  recentXpTransactions: XpTransactionResponse[]
}
